rules/react.js

```diff
@@ -9,47 +9,109 @@ module.exports = {
+        // Forbid vague propTypes
+        // 'react/forbid-prop-types'           : [2, {'forbid': ['any']}],
         // Enforce boolean attributes notation in JSX
         'react/jsx-boolean-value'           : [2, 'always'],
         // Enforce the closing bracket location for JSX multiline elements
         'react/jsx-closing-bracket-location': [2, {'selfClosing': 'after-props', 'nonEmpty': false}],
         // Disallow spaces inside of curly braces in JSX attributes
         'react/jsx-curly-spacing'           : [2, 'never'],
+        // Disallow spaces around equal signs in JSX attributes
+        'react/jsx-equals-spacing'          : [2],
+        // Ensure correct position of the first property
+        'react/jsx-first-prop-new-line'     : [2, 'multiline'],
+        // Enforce handler naming convention
+        'react/jsx-handler-names'           : [2],
         // Validate props indentation in JSX
         'react/jsx-indent-props'            : [2, 4],
+        // Validate JSX indentation
+        'react/jsx-indent'                  : [2, 4],
+        // Detect missing key prop
+        'react/jsx-key'                     : [2],
         // Limit maximum of props on a single line in JSX
         'react/jsx-max-props-per-line'      : [2, {'maximum': 3}],
+        // Prevent comments from being inserted as text nodes
+        'react/jsx-no-comment-textnodes'    : [2],
         // Prevent duplicate props in JSX
         'react/jsx-no-duplicate-props'      : [2],
+        // Prevent usage of unsafe target='_blank'
+        'react/jsx-no-target-blank'         : [2],
         // Disallow undeclared variables in JSX
         'react/jsx-no-undef'                : [2],
+        // Enforce PascalCase for user-defined JSX components
+        'react/jsx-pascal-case'             : [2],
+        // Enforce props alphabetical sorting
+        'react/jsx-sort-props'              : [2, {'callbacksLast': true, 'ignoreCase': true}],
+        // Validate spacing before closing bracket in JSX
+        'react/jsx-space-before-closing'    : [2],
+        // Validate whitespace in and around the JSX opening and closing brackets
+        'react/jsx-tag-spacing'             : [2],
         // Prevent React to be incorrectly marked as unused
         'react/jsx-uses-react'              : [2],
+        // Prevent variables used in JSX to be incorrectly marked as unused
+        'react/jsx-uses-vars'               : [2],
+        // Prevent missing parentheses around multilines JSX
+        'react/jsx-wrap-multilines'         : [2],
+        // Prevent usage of Array index in keys
+        'react/no-array-index-key'          : [2],
+        // Prevent passing of children as props
+        'react/no-children-prop'            : [2],
+        // Prevent problem with children and props.dangerouslySetInnerHTML
+        'react/no-danger-with-children'     : [2],
+        // Prevent usage of dangerous JSX properties
+        'react/no-danger'                   : [2],
```

Comment:
__*Question*__
I know we do this with some server side generated html. Is the plan to just ignore it in those cases?